eth/tracers/api.go

```diff
@@ -896,10 +896,11 @@ func (api *API) TraceCall(ctx context.Context, args ethapi.TransactionArgs, bloc
 	var traceConfig *TraceConfig
 	if config != nil {
 		traceConfig = &TraceConfig{
-			Config:  config.Config,
```

Comment:
Good catch thanks. But this whole block seems unnecessary. We can pass `config` directly to `traceTx`. `traceTx` does nil check and instantiates an empty object if necessary. Can you simply remove this couple of lines?

Reply:
They do not have same type (`traceTx` takes a `TraceConfig` and `TraceCall` takes a `TraceCallConfig`). So something like this?

I believe `Reexec` and `Override` were also mishandled (in the wrong object).

Also I'm not 100% sure about my line 873 (pretty new to Go)